src/main/java/com/e_gineering/TagMasterMojo.java

```diff
@@ -0,0 +1,71 @@
+package com.e_gineering;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Scm;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.*;
+
+/**
+ * Invokes configures the builds SCM settings based on environment variables from Jenkins, and does an scm:tag
+ * If the env.GIT_BRANCH matches the masterBranchPattern.
+ */
+@Mojo(name = "tag-master", defaultPhase = LifecyclePhase.VERIFY)
+public class TagMasterMojo extends AbstractGitEnforcerMojo {
+
+    @Parameter(defaultValue = "${project.version}", property = "tag", required = true)
+    private String tag;
+
+    @Component
+    private MavenSession mavenSession;
+
+    @Component
+    private BuildPluginManager pluginManager;
+
+    public void execute() throws MojoFailureException, MojoExecutionException {
+        String gitBranch = System.getenv("GIT_BRANCH");
+        String gitURL = System.getenv("GIT_URL");
+
+        if (gitBranch != null && gitURL != null) {
+            getLog().info("Detected GIT_BRANCH: '" + gitBranch + "' in build environment.");
+            getLog().info("Detected GIT_URL: '" + gitURL + "' in build enviornment.");
+
+            if (gitBranch.matches(masterBranchPattern)) {
+                getLog().info("Attempting to invoke an scm:tag execution for a build on the master branch...");
+
+                if (project.getScm() == null) {
+                    project.setScm(new Scm());
+                }
+                project.getScm().setDeveloperConnection("scm:git:" + gitURL);
```

Comment:
So this project is intended to help organizations with lots of builds reduce the amount of 'fiddling' they need to do with maven poms, and greatly simplify jenkins setup.

The assumption is that there is _not_ an SCM configuration. For public projects and things going into central, it would make a lot of sense to set one up. For corporate environments, or private forks in a corporate environment, it's far less beneficial to have the original scm stuff stashed into this pom. It also becomes problematic if someone declares HTTPS for the git transport, but the jenkins box is using SSH to clone the repo. You can end up in a bad situation where the pom.xml's scm connection and developerConnections cannot be authenticated because the local build box has no way to map the credentials.

So to simplify this entire thing, we pull the GIT information out of a Jenkins / Hudson environment (if it's there) and use that as the connection method to do the git tag.

And as a result, you don't _have_ to define any scm info on any project using this as a parent.